src/utils/number.js

```diff
@@ -48,8 +74,16 @@ export function isIntegerStr (str) {
  * @returns {'number' | 'BigNumber' | 'bigint' | 'Fraction'}
  */
 export function safeNumberType (numberStr, config) {
-  if (config.number === 'bigint' && !isIntegerStr(numberStr)) {
-    return config.numberFallback
+  if (config.number === 'bigint') {
```

Comment:
The following definition of safeNumberType passes all tests:
```
export function safeNumberType (numberStr, config) {
  if (config.number === 'bigint') {
    try {
      BigInt(numberStr)
    } catch {
      return config.numberFallback
    }
  }

  return config.number
}
```

If we adopted this, we would not need the three new "isXXXStr" functions, and we would be future-proofing mathjs against any other integer notations. Would you like to switch to this? If so, will you do the refactoring or would you like me to? Just let me know.

Reply:
That is a smart idea, I like it.

Often, this helper function `safeNumberType` is used in combination with `numeric`, like:

```js
res = numeric(res, safeNumberType(res, config))
```

In case of a `BigInt` this means the number is parsed twice (once to test the type, once to really parse it). If this turns out to be a performance issue we can look into optimizing this.